Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -74,13 +127,40 @@ void HTMLScriptElement::prepare_script(Badge<HTMLDocumentParser>)
     }
 
     auto source_text = child_text_content();
-    if (!has_attribute(HTML::AttributeNames::src) && source_text.is_empty())
+    if (!has_attribute(HTML::AttributeNames::src) && source_text.is_empty()) {
+        dbgln("HTMLScriptElement: Refusing to run empty script.");
         return;
+    }
 
-    if (!is_connected())
+    if (!is_connected()) {
+        dbgln("HTMLScriptElement: Refusing to run script because the element is not connected.");
         return;
+    }
 
-    // FIXME: Check the "type" and "language" attributes
+    String script_block_type;
+    bool has_type = has_attribute(HTML::AttributeNames::type);
+    bool has_language = has_attribute(HTML::AttributeNames::language);
+    if ((has_type && attribute(HTML::AttributeNames::type).is_empty())
+        || (!has_type && has_language && attribute(HTML::AttributeNames::language).is_empty())
+        || (!has_type && !has_language)) {
+        script_block_type = "text/javascript";
+    } else if (has_type) {
+        script_block_type = attribute(HTML::AttributeNames::type).trim_whitespace();
+    } else if (!attribute(HTML::AttributeNames::language).is_empty()) {
+        StringBuilder builder;
+        builder.append("text/");
+        builder.append(attribute(HTML::AttributeNames::language));
+        script_block_type = builder.to_string();
```

Comment:
```suggestion
        script_block_type = String::formatted("text/{}", attribute(HTML::AttributeNames::language));
```

No need for a string builder here IMO.